components/builder-core/src/access_token.rs

```diff
@@ -0,0 +1,92 @@
+// Copyright (c) 2017 Chef Software Inc. and/or applicable contributors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::path::PathBuf;
+use chrono::Utc;
+
+use hab_net::privilege::FeatureFlags;
+use error::{Error, Result};
+use integrations::{encrypt, decrypt};
+use protocol::{message, sessionsrv};
+
+pub const BUILDER_ACCOUNT_ID: u64 = 0;
+const ACCESS_TOKEN_PREFIX: &'static str = "TOKN";
+
+const BUILDER_TOKEN_EXPIRATION_SECONDS: i64 = 7200; // 2 hours
+const USER_TOKEN_EXPIRATION_SECONDS: i64 = 315360000; // 10 years
+
+pub fn generate_bldr_token(key_dir: &PathBuf) -> Result<String> {
```

Comment:
Considering this is used only one place, why not omit the constant, which allows the use of the more evocative `Duration` type? For example:

```rust
    let builder_token_lifetime = chrono::Duration::hours(2);

    generate_access_token(
        …
        builder_token_lifetime.num_seconds(),
    )
```

This may seem like a trivial thing, but I started thinking about it when I noticed `7200; // 2 hours`. I've dealt with code like this in the past where the value gets changed but the comment doesn't, leading to confusion.